src/vm/eventpipefile.h

```diff
@@ -25,37 +26,44 @@ class EventPipeFile : public FastSerializableObject
         );
         ~EventPipeFile();
 
-        // Write an event to the file.
         void WriteEvent(EventPipeEventInstance &instance);
 
-        // Serialize this object.
-        // Not supported - this is the entry object for the trace,
-        // which means that the contents hasn't yet been created.
-        void FastSerialize(FastSerializer *pSerializer)
+        void WriteEnd();
+
+        const char* GetTypeName()
         {
             LIMITED_METHOD_CONTRACT;
-            _ASSERTE(!"This function should never be called!");
+            return "Trace";
         }
 
-        // Get the type name of this object.
-        const char* GetTypeName()
+        void FastSerialize(FastSerializer *pSerializer)
```

Comment:
I know that you got rid of the support for forward references as part of this change to make it easier to support streaming.  How do you make sure that this block of information is extensible since the forward references are now gone?

Reply:
See https://github.com/Microsoft/perfview/blob/master/src/TraceEvent/EventPipe/EventPipeFormat.md#versioning-the-format-while-maintaining-compatibility